### ISSUE TITLE
Custom Components not loaded (cache issue?)

### ISSUE THREAD

**Issue by emandtf** [home-assistant/frontend#18728]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

If browsers have Cache Disabled all custom components (I'm using vertical-stack-in-card, mini-graph-card and multi-entity-row) won't be recognized until I remove "_Disable cache_" flag from Development Panel.

**Even the FIRST access** from a different browser (like on a new device) doesn't recognize custom components even if in the Console Log their initialization seems completed (see attached file).
![20231123_console_log](https://github.com/home-assistant/frontend/assets/13537385/9042acdf-3b0c-4a26-bb73-e1ddadd85db3)
Closing and re-opening tab/browser, after this FIRST visit with issues, seems let components working again.

It seems like custom components JS files are loaded just once and then retained in cache and served ONLY from it, so if it is disabled them won't work.
I think it's something related to _Service Worker_ and its adavanced caching system.

**PS**: After this test it was hard to let custom components work again (after cache enabled again) even doing normal page Refresh multiple times.
I have to close browser Tab and then open another one to have custom components working again.

### Describe the behavior you expected

Clearing browser cache or just do a login from a complete new device and accessing a Dashboard in whic there are some Custom Components present have to work fine without the need to Reload the page many times until them are recognized.

### Steps to reproduce the issue

1. Open Developer Panel (F12 on Firefox and Chrome)
2. Switch to Network tab
3. Enable "_Disable cache_" checkbox
4. Refresh a Lovelace dashboard where an custom component is placed, or just do CTRL+SHIFT+R to refresh the page WITHOUT using already cached files
5. Then all custom components are not recognized and won't work

![20231123_firefox_custom_component](https://github.com/home-assistant/frontend/assets/13537385/f4a73e6a-0ff4-4a43-ada6-164be9c0e5e0)

### What version of Home Assistant Core has the issue?

core-2023.11.2

### In which browser are you experiencing the issue with?

Firefox and Chrome 119.x

### Which operating system are you using to run this browser?

Windows 11 and Ubuntu 21

**Comment by karwosts** [comment]
I'll say I don't observe this. I have caching disabled and my custom cards load on reload.

Though I do use HTTP and not HTTPS, and I may have heard that the caching behavior is different between the two, something to do with https service workers. But I'm not that knowledgeable about that.

**Comment by emandtf** [comment]
I'm just "new" about Custom Components but I'm using HA from few years. Just yesterday I decided to implement two custom components and I've followed all guides to install them via HACS.
Seems worked fine just installed because the installing procedure ended with an "refresh" action and the components were ready to be used.

Today I tested my HA from remote (port forwarded and reversed Nginx proxy for SSL) with issues I reported before.

Now I'm at home and I'm using Companion App while connected to Wifi and the App uses a local "HTTP://192.168..:8123" (directly to HA without anu proxy) connection, and this is the result most of the times I open the App in the same page reported before:
![SmartSelect_20231123_200619_Home Assistant](https://github.com/home-assistant/frontend/assets/13537385/7493d12e-c302-47a4-9760-8f8f0c70b2e8)
Other times it loads correctly 🤷‍♂️

It's a bit frustrating.........
